tests/unit/source_code/samples/functional/pyspark/dataframe-write-insert-into.py

```diff
@@ -0,0 +1,13 @@
+# ucx[direct-filesystem-access:+1:0:+1:0] The use of direct filesystem references is deprecated: s3://bucket/path
+df = spark.read.csv("s3://bucket/path")
+for i in range(10):
+
+    ## Check a literal reference to a known table that is migrated.
+    # ucx[table-migrate:+1:0:+1:0] Table old.things is migrated to brand.new.stuff in Unity Catalog
+    df.write.insertInto("old.things")
+
+    ## Check a literal reference to an unknown table (that is not migrated); we expect no warning.
+    df.write.insertInto("table.we.know.nothing.about")
+
+    ## Check that a call with too many positional arguments is ignored as (presumably) something else; we expect no warning.
+    df.write.insertInto("old.things", None, "extra-argument")
```

Comment:
looks like a bug in UCX analysis

Reply:
As written, it's intentional behaviour (covered by the `Matcher` `max_arg` property). 

There are two issues with this behaviour that I've got in the back of my mind:

 - I'm not sure what we need to cope with in terms of future compatibility, but the Spark APIs tend to add new positional arguments over time with default values. The current `max_args` spec instantly means that future versions of these APIs won't match. This drift has already happened over the Spark 3.x evolution, and it maintains source/binary compatibility (on the Python side) so I would expect this to continue. As such I think it might be worth defaulting to treating methods as having an implicit `*args` even if they don't, with `max_args` limited to cases where it's needed to disambiguate or eliminate false positives.
 - The current matcher specs don't really cover the way python arguments can be specified. Some limitations include missing `*args` (workaround: max=1000) and `**kwargs` (no workaround) support. It also doesn't understand that all positional arguments have a name, and often these are used. (Style guides and linting rulesets often mandate using named syntax for int/boolean parameters, so I expect to find some quirks in this area that we need to address.)